### PR TITLE
Wizard dialogues part fix for #2704.

### PR DESCRIPTION
**Pull request by easmetz**
Part fix for #2704.

### cc
@Charlotte-Morgan please review

### CODE REVIEW

**Comment by Gustry** [comment]
I don't know why you got some commits from me in your PR. We can check that next week.

**Comment by Charlotte-Morgan** [comment]
I think @easmetz - really only wants the last 2 commits :)

**Comment by Charlotte-Morgan** [comment]
hi @easmetz - the text changes look fine. It would be great to add a citation element to the definitions and reference the souce when you have updated descriptions using UNISDR text.
@timlinux - will write a ticket for making the widget in the wizard scalable

@Gustry - can you please advise how to exclude the commits that are not new or relevant?  :)

**Comment by timlinux** [comment]
Looking at the changes I think we can just squash merge and not worry too much about @Gustry's changes being mixed in.

**Comment by easmetz** [comment]
@Gustry I don't understand why I have changes from you in my pull request, but I'm glad we can squash merge. 

In definitions.py, there are two definitions for hazard - "hazard"(https://github.com/inasafe/inasafe/blob/develop/safe/definitions.py#L40) and "hazards" (https://github.com/inasafe/inasafe/blob/develop/safe/definitions.py#L299) and two definitions for exposure - "exposure" (https://github.com/inasafe/inasafe/blob/develop/safe/definitions.py#L50) and "exposures" (https://github.com/inasafe/inasafe/blob/develop/safe/definitions.py#L369) . Why are there two definitions for these words? These two definitions (exposure and exposures) are not consistent - should I make them the same?
@Charlotte-Morgan I am currently looking at incorporating a reference in the definitions.

**Comment by easmetz** [comment]
I have now updated the definitions.py file to have references to UNISDR, but I still need to figure out what to do with the double definitions. Let me know if you think another way of referencing UNISDR is preferable.

File: safe/definitions.py [hunk 1]
```diff
@@ -47,6 +47,8 @@
         'property damage, loss of livelihoods and services, social and '
         'economic disruption, or environmental damage. For example; flood, '
         'earthquake, tsunami and volcano are all examples of hazards.')
+        # source UNISDR (2009) Terminology on disaster risk reduction.
```
Comment (timlinux) [inline safe/definitions.py]: I think it is better to do these as dictionary key / value pairs. If they are comments we cannot use these in reports etc. My suggestion is to follow the style of the impact function metadata (which allows 0 or more citations. Take a look at this example:

```
            'citations': [
                tr('Sengara, W., Suarjana, M., Yulman, M.A., Ghasemi, H., and '
                   'Ryu, H. (2015). '
                   'An empirical fatality model for Indonesia based on '
                   'a Bayesian approach. Submitted for Journal of the '
                   'Geological Society')  # FIXME
            ],
``` 

(taken from ITB impact function' metadata.

File: safe/definitions.py [hunk 2]
```diff
@@ -55,6 +57,8 @@
         'An <b>exposure</b> layer represents people, property, systems, or '
         'other elements present in hazard zones that are thereby subject to '
```
Comment (timlinux) [inline safe/definitions.py]: Same comment above regarding citation as data not as comment...

**Comment by timlinux** [comment]
Hi @easmetz 

> In definitions.py, there are two definitions for hazard - "hazard"(https://github.com/inasafe/inasafe/blob/develop/safe/definitions.py#L40) and "hazards" (https://github.com/inasafe/inasafe/blob/develop/safe/definitions.py#L299) and two definitions for exposure - "exposure" (https://github.com/inasafe/inasafe/blob/develop/safe/definitions.py#L50) and "exposures" (https://github.com/inasafe/inasafe/blob/develop/safe/definitions.py#L369) .

> Why are there two definitions for these words? The definitions for each word set (eg exposure and exposures, hazard and hazards) are not consistent - should I make them the same?

Please don't remove them - they are different things e.g.:

```
layer_purpose_hazard = {
    'key': 'hazard',
    'name': tr('Hazard'),
    'description': tr(
        'A <b>hazard</b> layer represents '
        'something that will impact on the people, infrastructure or  '
        'land cover in an area. For example; flood, earthquake, tsunami and '
        'volcano are all examples of hazards.')
}
```

this describes a layer's purpose...

```
# Renamed key from hazard to hazards in 3.2 because key was not unique TS
hazards = {
    'key': 'hazards',
    'name': tr('Hazards'),
    'description': tr(
        '<b>Hazards</b> (also called disasters) are what we call the data '
        'layers that describe the extent and magnitude of natural events '
        '(such as earthquakes, tsunamis and volcanic eruptions) that could '
        'potentially cause an event or series of events that threaten and '
        'disrupt the lives and livelihoods of people.'),
    'types': hazard_all
}
```

... this describes all the different types of hazards (note the additional ``types`` attribute). 

Well spotted though! I think we could at least normalise the descriptions so that there is a canonical description that can be maintained in one place...I will do that in a separate PR.

**Comment by timlinux** [comment]
@easmetz I see we have some conflicts on this PR. I will fix these and merge later today unless you tell me to do otherwise. You can always make follow up PR's if you want to do more tweaking - OK?

**Comment by easmetz** [comment]
Thanks @timlinux. That sounds good to me - I don't have any extra time to do it today but I should later in the week.

**Comment by timlinux** [comment]
@easmetz ok perfecto! :-)

**Comment by timlinux** [comment]
Closing this as per the discussion above - dealt with in #3046

**Pull request closed without merging**